src/vs/workbench/contrib/terminal/browser/terminalTab.ts

```diff
@@ -297,14 +296,14 @@ export class TerminalTab extends Disposable implements ITerminalTab {
 		return this._terminalInstances[this._activeInstanceIndex];
 	}
 
-	public getLayoutInfo(isActive: boolean): ITerminalTabLayoutInfoById {
+	public getLayoutInfo(isActive: boolean, isRemote: boolean): ITerminalTabLayoutInfoById {
 		const isHorizontal = this.splitPaneContainer?.orientation === Orientation.HORIZONTAL;
-		const remoteInstances = this.terminalInstances.filter(instance => typeof instance.remoteTerminalId === 'number');
-		const totalSize = remoteInstances.map(instance => isHorizontal ? instance.cols : instance.rows).reduce((totalValue, currentValue) => totalValue + currentValue, 0);
+		const instances = isRemote ? this.terminalInstances.filter(instance => typeof instance.remoteTerminalId === 'number') : this.terminalInstances;
+		const totalSize = instances.map(instance => isHorizontal ? instance.cols : instance.rows).reduce((totalValue, currentValue) => totalValue + currentValue, 0);
 		return {
 			isActive: isActive,
-			activeTerminalProcessId: this.activeInstance?.processId || 0,
-			terminals: remoteInstances.map(t => {
+			activeTerminalProcessId: isRemote ? this.activeInstance?.processId || 0 : this.activeInstance?.id || 0,
```

Comment:
Should we be throwing or setting it to undefined instead if `activeInstance` doesn't exist?

Reply:
right now the type is a number. should I change that to number | undefined or set to zero in the case that activeInstance doesn't exist?